# Checkpoint 9 answers/solutions

## Learning Target 2

- Addition: 0101110110
- Subtraction: 01110000
- Multiplication: 0110111
- Division: 1010 with remainder 100


---

## Learning Target 3

>If the store is closed, I will come back tomorrow

1. The store is closed
2. I will come back tomorrow
3. If I come back tomorrrow, then the store is closed 
4. If I do not come back tomorrow, then the store is not closed
5. If the store is not closed, I will not come back tomorrow
6. The store is closed, but I won't come back tomorrow. 

### Common errors: 

- Writing the converse as "*I will come back tomorrow if the store is closed*". This is not a converse, it's the same sentence as the original just written in a different order. **Tip: Do not include the word "if" when you write the hypothesis.** (That is, "If the store is closed" is not fully correct -- leave "if" off.) Doing so might be causing you to include the word "if" when you relocate the hypothesis to the end of the sentence, but the result is the same as the original.
- Writing the negation as "*The store is closed, I won't come back tomorrow*" using a comma instead of the word "and" (or "but"). **The comma is ambiguous and should not be used.** The comma could mean "and", or it could mean "or", or it could mean "therefore", or "because". All four possible interpretations are logically different. **Just use "and".** 


---

## Learning Target 6

1.
> $a_0 = 2, a_1 = 1$, and if $n >1$ $a_n = a_{n-1} + a_{n-2}$

In [2]:
# Code to generate the results 

def a(n):
    if n == 0: return 2
    elif n == 1: return 1
    else: return a(n-1) + a(n-2)


[a(n) for n in range(2,6)]

[3, 4, 7, 11]

2.
> 3, 8, 13, 18, 23, ...

$a_0 = 3$, and for $n > 0$ $a_n = a_{n-1} + 5$

---

## Learning Target 7

> For every integer $n \geq 1$, $1 + 2 + 3 + \cdots + n = \dfrac{n(n+1)}{2}$. 

1. The predicate is $P(n): 1 + 2 + 3 + \cdots + n = \dfrac{n(n+1)}{2}$
2. The base case is when $n = 1$ 
3. To show the base case holds, plug in $n = 1$ on the left and $n = 1$ on the right and let's see if they are equal. On the left, $n=1$ would just produce a single-term "sum", the number $1$. On the right, $n=1$ would give $\frac{1(2)}{2} = 1$. These are equal, so the base case holds. 
4. Assume that $1 + 2 + 3 + \cdots + k = \dfrac{k(k+1)}{2}$ for some $k \geq 1$. 
5. Prove that $1 + 2 + 3 + \cdots + (k+1) = \dfrac{(k+1)(k+1+1)}{2}$. 


### Common errors

- Including the quantifier "For every integer $n \geq 1$" in the predicate statement. **Including the quantifier is specifically prohibited in the success criteria written underneath the problem!** This is because if you put a quantifier on this, it's not a predicate. 
- In part 3, showing that the right side of the equation in the predicate equals 1 without talking about why the left side also equals 1. It's not enough just to plug $1$ into $\frac{n(n+1)}{2}$ and get a correct answer, **you also must explain why the left side gives the same result**. 
- Failure to use parentheses in step 5, that is, writing the fraction as $\dfrac{k+1(k+1+1)}{2}$. The parentheses are necessary to make the algebra expression correct, they are not merely decorative!
- **Rushing through the response and not writing things out in words.** Don't try to abbreviate everything with mathematical notation -- you should be using English words to explain things that need to be explained. Specific problems this is causing: 
  - Writing only disconnected equations in part 3. For example giving a response that consists only of: $\frac{1 \times 2}{2} = 1$. By itself this means nothing. You are asked to **explain why the predicate is true in the base case** here. Just writing a fraction isn't enough. 
  - Not writing the word "assume" in part 4, but just writing the equation. **This is a critical error** because the reader needs to know what you intend to do here. 
  - Not writing the word "prove" in part 5, also a critical error and for the same reason. 

---

## Learning Target 8

1.
   (a) $\{14, 23\}$ 
   (b) $\{0,1\}$
   (c) Incorrect syntax (because it gives a set followed by a function, not followed by a predicate)
2. More than one possible right answer here. One example: 
   $\{letter \in \{a,b,c,\dots, z\} \, : \, letter \, \text{comes after} \, w\}$

### Common errors

- In 1(a), either missing the numbers 14 or 23; or going too far and including numbers such as 32, 41, and 50. The set builder notation states that $x$ only ranges between 1 and 30. 
- Giving $\{0,1,0,1,0,1,\dots\}$ as the response to 1(b). This is not a valid set because it contains duplicate elements. 
- Interpreting $x \, \% \, 2$ as "$x$ divided by 2" in 1(b) resulting in the set $\{0,1/2,1,3/2, \dots\}$. Not sure where that's coming from but it happened more than once. **Make sure to read the notation carefully!** "$x \, \% \, 2$" and "$x \div 2$" mean different things. 
- Giving the following as a response to part 2: $\{a \, : a \in \{x,y,z\}\}$ this is correct but trivial and does not demonstrate skill on the Learning Target. That's why **this is explicitly ruled out in the Success Criteria printed below the problem statement.**
- Using $x$ as the name of the variable that belongs to a set containing the letter "x". For example $\{x \in \{a,b,c,\dots,x,y, z\} \, : \, x \, \text{comes after} \, w\}$. This doesn't work, because "$x \in \{a,b,c,\dots,x,y, z\}$" is merely a true statement, not a set of elements that we will then filter with a predicate. It also doesn't work because "$x \, \text{comes after} \, w$" is simply "false", it's not a predicate we can use to filter elements out of a set. **Tip: Just as when writing computer code, avoid using the same variable name in two different contexts.** 

## Learning Target 9

>$A = \{3, 4, 5, 6\}$, $B = \{2, 4, 6, 8\}$, $C = \{0, 5\}$ and   $U = \{0, 1, 2, \dots, 10\}$

1. $\{5\}$
2. $\{0,2,4,5,6,8\}$
3. $\{2,8\}$
4. $\{0,1,2,7,8,9,10\}$
5. $\{2,4,6,8\}$ or just $B$
6. $\emptyset$ (any set intersected with $\emptyset$ is $\emptyset$, no computation needed)
7. $\{\emptyset, \{0\}, \{5\}, \{0,5\}\}$
8. 8

### Common errors 

- Not including the empty set in the power set ${\cal{P}}(C)$. The empty set is a subset of every set, so it is always included in the power set of a set. 
- Not computing cardinality in part 8 but instead giving $B \times C$ (which is a set) instead of $|B \times C|$ (which is a number). 
- Two errors which are not deal-breakers but could cause problems: 
  - Not using set brackets around sets, for example giving the answer for part 2 as "$(0,2,4,5,6,8)$" (using parentheses instead of braces) or as just "$0,2,4,5,6,8$" (as a sequence with no delimiters at all). **Make sure to use correct set notation.**
  - Putting set braces around the empty set, like: $\{\emptyset\}$. This is not the empty set; it is a set with one element, namely $\emptyset$. 

## Learning Target 12

1. Four slots and 36 choices per slot: 

In [3]:
36**4

1679616

2. First three slots have 36 choices each, last two have just 2 options

In [4]:
36**3 * 2

93312

3. To start with $0$ leaves three slots with 36 choices each. To end with an "o" does the same. **And, we need to count the overlap:** If a password starts with $0$ and ends in "o" then there are two slots with 36 choices each: 

In [5]:
starts_with_zero  = 36**3
ends_with_o = 36**3
overlap = 36**2

starts_with_zero + ends_with_o - overlap

92016

### Common errors 

- By far the most common error is **not subtracting out the overlap in the two cases**. Whenever finding the number of things that satisfy an "either/or" condition, you'll *always* need to find the number that satisfy "both/and" and apply the **Principle of Inclusion and Exclusion**. 
- In part 3, finding the number of passwords that *both* start with 0 *and* end in "o" -- and returning that number ($36^2$) as the answer. It's *involved* in finding the count, but you're asked to count the number that *either* start with 0 *or*  end in "o". 

## Learning Target 13

In [9]:
from math import factorial

# 1(a)
print("6! = ", factorial(6))

# 1(b)

print("11 choose 6 = ", (factorial(11)//(factorial(6) * factorial(5))))

6! =  720
11 choose 6 =  462


2(a). Five books on a shelf, number of arrangements is $5!$

In [10]:
factorial(5)

120

2(b). Selecting four books from 10 and then rearranging: Number of arrangements is $P(10,4) = \frac{10!}{6!}$ (*not* $\binom{10}{4}$)

In [11]:
factorial(10)//factorial(6)

5040

## Common errors 

- By far the most common is **using the binomial coefficient on 2(b)** and computing the number as $\binom{10}{4} = 210$. This number, 210, gives the number of ways to select 4 books at random from the group of 10. But this is not the counting situation we're in: We are choosing 4 books from 10 and then putting them on the shelf "in a specific order" (as it says in the statement). This is a "4-permutation". Or consider it as a [license plate problem](https://publish.obsidian.md/mth225/Combinatorics/License+plate+problem): There are four spots on the bookshelf to fill. There are 10 ways to fill the first slot; then 9 ways to fill the second since the first was already filled; then 8 ways to fill the third; and 7 ways to fill the fourth. Since the choices are in sequence, the total count is $10 \cdot 9 \cdot 8 \cdot 7 = 5040$. Note this is the same as $P(10,4)$; in fact, in class, we derived the formula for $P(n,k)$ by thinking about license plate problems. 